src/lando/api/tests/test_phabricator_client.py

```diff
@@ -83,3 +85,24 @@ def test_phabricator_exception(get_phab_client):
             phab.call_conduit("differential.query", ids=["1"])[0]
         assert e_info.value.error_code == error["error_code"]
         assert e_info.value.error_info == error["error_info"]
+
+
+def test_phabricator__get_conduit_result_data(get_phab_client, monkeypatch):
+    phab = get_phab_client(api_key="api-key")
+    mock_call_conduit = mock.MagicMock()
+
+    first_batch = []
+    second_batch = []
+    for i in range(100):
+        first_batch.append(f"batch-1-{i}")
+        second_batch.append(f"batch-2-{i}")
+
+    initial_result = {"data": first_batch.copy(), "cursor": {"after": 1234}}
+    second_result = {"data": second_batch.copy(), "cursor": {}}
+
+    mock_call_conduit.side_effect = [initial_result, second_result]
```

Comment:
Optimally we would use the `PhabricatorDouble` to create 2 pages of `differential.diff.search` (or some other endpoint) results and ensure the new retrieval method correctly returns all the results. This seems reasonable for now, but we should do that as a follow-up.

Reply:
I think increasing the integration reach of this test could be beneficial, but don't think it's necessary since we are not interacting directly with the Phabricator API or care about the internals of it, instead relying on another instance method that has an expected response.

I think if we wanted to test this via PhabricatorDouble, we should add a test elsewhere (maybe in `test_revisions`) that does a more proper end-to-end test by creating revisions, diffs, etc. and then calling the API. As a side note, it looks like the only endpoint that properly supports this in the mocked `PhabricatorDouble` class is `transaction_search` so this wouldn't work if we tried it today.